Gem/Code/Source/Components/UI/UiPlayerArmorComponent.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Serialization/EditContext.h>
+#include <LyShine/Bus/UiImageBus.h>
+#include <LyShine/Bus/UiTextBus.h>
+#include <Source/Components/UI/UiPlayerArmorComponent.h>
+
+namespace MultiplayerSample
+{
+    void UiPlayerArmorComponent::Activate()
+    {
+        UiPlayerArmorNotificationBus::Handler::BusConnect();
+    }
+
+    void UiPlayerArmorComponent::Deactivate()
+    {
+        UiPlayerArmorNotificationBus::Handler::BusDisconnect();
+    }
+
+    void UiPlayerArmorComponent::OnPlayerArmorChanged(float armorPointsForLocalPlayer, float startingArmor)
+    {
+        const AZStd::string armorTextValue = AZStd::string::format("%.0f", armorPointsForLocalPlayer);
+        UiTextBus::Event(m_armorText, &UiTextBus::Events::SetText, armorTextValue.c_str());
+
+        if (startingArmor > 1.f)
+        {
+            const float fillAmount = AZStd::clamp(armorPointsForLocalPlayer / startingArmor, 0.f, 1.f);
+            UiImageBus::Event(m_armorVisualEntity, &UiImageBus::Events::SetFillAmount, fillAmount);
```

Comment:
it would be cool if the fill volume went red or something if it falls below 20% 😉